docker-api/src/main/java/com/yahoo/vespa/hosted/dockerapi/ContainerResources.java

```diff
@@ -1,26 +1,51 @@
 // Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 package com.yahoo.vespa.hosted.dockerapi;
 
+import java.util.Objects;
+
 /**
  * @author valerijf
  */
 public class ContainerResources {
-    public static final ContainerResources UNLIMITED = ContainerResources.from(0, 0);
+    public static final ContainerResources UNLIMITED = ContainerResources.from(0, 0, 0);
+    private static final int CPU_PERIOD = 100_000; // 100 µs
 
+    private final double cpus;
     private final int cpuShares;
     private final long memoryBytes;
 
-    ContainerResources(int cpuShares, long memoryBytes) {
+    ContainerResources(double cpus, int cpuShares, long memoryBytes) {
+        this.cpus = cpus;
         this.cpuShares = cpuShares;
         this.memoryBytes = memoryBytes;
+
+        if (cpus < 0)
+            throw new IllegalArgumentException("CPUs must be a positive number or 0 for unlimited, was " + cpus);
+        if (cpuShares < 0)
+            throw new IllegalArgumentException("CPU shares must be a positive integer or 0 for unlimited, was " + cpuShares);
+        if (memoryBytes < 0)
+            throw new IllegalArgumentException("memoryBytes must be a positive integer or 0 for unlimited, was " + memoryBytes);
     }
 
-    public static ContainerResources from(double cpuCores, double memoryGb) {
+    public static ContainerResources from(double cpus, double cpuCores, double memoryGb) {
         return new ContainerResources(
+                cpus,
                 (int) Math.round(10 * cpuCores),
                 (long) ((1L << 30) * memoryGb));
     }
 
+    public double cpus() {
```

Comment:
Could you document somewhere the effect of setting both cpu shares and cpu quota/period? Do we get the minimum of what both would have limited if they were evaluated independently?

Reply:
Not sure where to best put, but added some comments on the variables